source/server/active_tcp_listener.h

```diff
@@ -71,48 +75,45 @@ class ActiveTcpListener final : public Network::TcpListenerCallbacks,
                       bool hand_off_restored_destination_connections, bool rebalanced) override;
 
   /**
-   * Remove and destroy an active connection.
-   * @param connection supplies the connection to remove.
+   * Create active connection from the server connection.
    */
-  void removeConnection(ActiveTcpConnection& connection);
+  void newActiveConnection(const Network::FilterChain& filter_chain,
+                           Network::ServerConnectionPtr server_conn_ptr,
+                           std::unique_ptr<StreamInfo::StreamInfo> stream_info) override;
 
   /**
-   * Create a new connection from a socket accepted by the listener.
+   * Return the active connections container attached with the given filter chain.
    */
-  void newConnection(Network::ConnectionSocketPtr&& socket,
-                     std::unique_ptr<StreamInfo::StreamInfo> stream_info);
+  ActiveConnections& getOrCreateActiveConnections(const Network::FilterChain& filter_chain);
 
   /**
-   * Return the active connections container attached with the given filter chain.
+   * Update the listener config. The follow up connections will see the new config. The existing
+   * connections are not impacted.
    */
-  ActiveConnections& getOrCreateActiveConnections(const Network::FilterChain& filter_chain);
+  void updateListenerConfig(Network::ListenerConfig& config);
 
   /**
-   * Schedule to remove and destroy the active connections which are not tracked by listener
-   * config. Caution: The connection are not destroyed yet when function returns.
+   * Schedule to remove and destroy the active connection owned by the filter chain.
    */
-  void
-  deferredRemoveFilterChains(const std::list<const Network::FilterChain*>& draining_filter_chains);
+  void deferRemoveFilterChain(const Network::FilterChain* filter_chain) override;
 
   /**
-   * Update the listener config. The follow up connections will see the new config. The existing
-   * connections are not impacted.
+   * Remove and destroy an active connection.
+   * @param connection supplies the connection to remove.
    */
-  void updateListenerConfig(Network::ListenerConfig& config);
+  void removeConnection(ActiveTcpConnection& connection);
 
-  Network::TcpConnectionHandler& parent_;
-  Network::ListenerPtr listener_;
-  const std::chrono::milliseconds listener_filters_timeout_;
-  const bool continue_on_listener_filters_timeout_;
-  std::list<ActiveTcpSocketPtr> sockets_;
-  absl::flat_hash_map<const Network::FilterChain*, ActiveConnectionsPtr> connections_by_context_;
+  absl::node_hash_map<const Network::FilterChain*, std::unique_ptr<ActiveConnections>>
```

Comment:
Why was this changed from flat to node hash map?

Reply:
Hmm, this should work, what is the compile error you get?

Reply:
OK... I don't see error in this PR right now. 
Revert to `flat_hash_map`. If I see it in the next PR we can discuss with concrete error.

Reply:
I had the illusion it is done.

Now I am trying to move to protected and I realize it was public even before this PR.

This things I am doing is:
1. declare it protected.
2. add const getter. it is for test only, I can take a look how to avoid using it in the test.
3. add a remove method in this class to mutate this list.